IdentityCore/src/oauth2/account/MSIDAccount.m

```diff
@@ -178,4 +178,68 @@ - (NSString *)description
     return [NSString stringWithFormat:@"MSIDAccount environment: %@ storage environment %@ realm: %@ username: %@ homeAccountId: %@ accountType: %@ localAccountId: %@", self.environment, self.storageEnvironment,  self.realm, self.username, self.accountIdentifier.homeAccountId, [MSIDAccountTypeHelpers accountTypeAsString:self.accountType], self.localAccountId];
 }
 
+#pragma mark - MSIDJsonSerializable
+
+- (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)error
+{
+    if (!(self = [super init]))
+    {
+        return nil;
+    }
+    
+    if (![json msidAssertType:NSDictionary.class ofKey:@"account_identifier" required:YES error:error])
+    {
+        return nil;
+    }
+    
+    self.accountIdentifier = [[MSIDAccountIdentifier alloc] initWithJSONDictionary:json[@"account_identifier"] error:error];
```

Comment:
I thought nested dictionaries weren't allowed. @antrix1989?

Reply:
correct, we need to use flat json.

Reply:
addressed :)